lib/puppet_https.rb

```diff
@@ -20,19 +23,19 @@ def make_ssl_request(url, req)
     connection.read_timeout = @settings['read_timeout'] || 90 #A default timeout value in seconds
 
     connection.verify_mode = OpenSSL::SSL::VERIFY_PEER
-    ca_file = @settings['ca_certificate_path']
-    certpath = @settings['certificate_path']
+    ca_file   = @settings['ca_certificate_path']
+    certpath  = @settings['certificate_path']
     pkey_path = @settings['private_key_path']
 
-    if File.exists?(ca_file)
+    if ca_file and File.exists?(ca_file)
```

Comment:
Hmm, is this change to prevent an exception being thrown if the file paths weren't provided (i.e. it tries to do `File.exists?(nil)`)? If so I think this if statement will still throw an exception:

```ruby
[4] pry(main)> if "hello" and File.exists?(nil)
[4] pry(main)*   puts "nope"
[4] pry(main)* end
(pry):6: warning: string literal in condition
TypeError: no implicit conversion of nil into String
from (pry):4:in `exists?'
```

It might be better to set them to empty strings above as a default like so:

```ruby
ca_file = @settings['ca_certificate_path'] ||= ""
```

Then `File.exists?` will evaluate an empty string rather than nil.